site/src/lib/blocks.ts

```diff
@@ -84,49 +85,56 @@ const getRepositoryUrl = (
 export const readBlocksFromDisk = async (): Promise<
   ExpandedBlockMetadata[]
 > => {
-  return glob
-    .sync(`${process.cwd()}/public/blocks/**/block-metadata.json`)
-    .map((path: string): ExpandedBlockMetadata => {
-      const packagePath = path.split("/").slice(-3, -1).join("/");
-
-      const metadata: ExpandedBlockMetadata = {
-        // @todo should be redundant to block's package.json#name
-        componentId: `${FRONTEND_URL}/blocks/${packagePath}`,
-        packagePath,
-        ...JSON.parse(fs.readFileSync(path, { encoding: "utf8" })),
-      };
-
-      const storedBlockInfo: StoredBlockInfo = JSON.parse(
-        fs.readFileSync(`${process.cwd()}/../hub/${packagePath}.json`, {
-          encoding: "utf8",
-        }),
-      );
-
-      const repository = getRepositoryUrl(
-        metadata.repository ?? storedBlockInfo.repository,
-        storedBlockInfo.commit,
-      )?.replace(/\/$/, "");
-
-      return {
-        ...metadata,
-        author: metadata.packagePath.split("/")[0]!.replace(/^@/, ""),
-        icon: generateBlockFileUrl(metadata.icon, metadata.packagePath),
-        image: generateBlockFileUrl(metadata.image, metadata.packagePath),
-        source: generateBlockFileUrl(metadata.source, metadata.packagePath)!,
-        variants: metadata.variants?.length
-          ? metadata.variants?.map((variant) => ({
-              ...variant,
-              icon: generateBlockFileUrl(variant.icon, metadata.packagePath)!,
-            }))
-          : null,
-        schema: generateBlockFileUrl(metadata.schema, metadata.packagePath)!,
-        repository,
-        blockPackagePath: `/${metadata.packagePath
-          .split("/")
-          .join("/blocks/")}`,
-        lastUpdated: null, // TODO: derive from block data when provided by the hub
-      };
+  const blockMetadataFilePaths = await globby(
+    path.resolve(process.cwd(), `public/blocks/**/block-metadata.json`),
+  );
+
+  const result: ExpandedBlockMetadata[] = [];
+  for (const blockMetadataFilePath of blockMetadataFilePaths) {
```

Comment:
Most of the diff is related to whitespace. I switched from `.map()` to `for of` to be able to use `await` in the loop.